Loosen requirement for a major version bump (#2510)
open-telemetry/opentelemetry-specification: specification/versioning-and-stability.md

### Major version bump

> Major version bumps MUST occur when there is a breaking change to a stable interface, the removal of a deprecated signal.

Reviewer: Specifically for the .NET case... (dont know if this is similar for other languages..)
we can leave this part of spec as-is, and add a clarification/exception to this: if the underlying language/runtime version itself is going out-of-support, then Major version bump need not be done...
(The key reason OTel .NET dropped support for an underlying runtime is because that runtime version itself is unsupported.)

Author: All languages I "know" (.NET, Go, Python, Java) have some policies regarding version support.

I am open to any proposal to clarify it. I have currently no idea how to phrase it in a clear way that will not bring any potential confusion.

Author: It looks like OTel Python has a similar policy https://github.com/open-telemetry/opentelemetry-python/issues/2360

@open-telemetry/python-maintainers 

Similarly for OTel Go: https://github.com/open-telemetry/opentelemetry-go/pull/2679

Reviewer 2: I support removing this clause. I don't see why it is necessary to increment Otel major version number because we no longer support an old version of a runtime.

@tedsuo you wrote this originally, maybe you can help clarify what was the idea.

Reviewer 3: I think because a user who tests with some version expects to be able to accept minor version increases without their application breaking. If they are on a runtime which loses support, this expectation is violated.

Reviewer 3: I agree different language users have very different expectations. The definition of "support" can also be different language-to-language. For example in some languages it may be impossible to even compile with an older compiler, but in another language it may simply mean it is simply not tested with that runtime.

Author: > Perhaps the wording should not be simply removed, but clarified that "may or may not require a major bump, see specific SIG guidelines/policy"

Proposal: 

```md
### Language or runtime version support

Each language implementation SHOULD define how the removal of a supported language/runtime version
affects its versioning. As a rule of thumb, it SHOULD take the recommended approach in a given ecosystem. 
```

PS. I find it hard to read. Edits are more than welcome.

Reviewer 4: lgtm

Reviewer 5: Yes, definitely. I agree with removing this part of the spec, major version bumps should happen when breaking API changes happen not when a specific version of the interpreter (or equivalent in other languages) happens.

Author: Addressed https://github.com/open-telemetry/opentelemetry-specification/pull/2510/commits/b13196b6616fab4369ca1255ac7c504894050445

### Major version bump

> Major version bumps MUST occur when there is a breaking change to a stable interface, the removal of a deprecated signal.

Reviewer 4: ```suggestion
Major version bumps MUST occur when there is a breaking change to a stable interface or a deprecated signal is removed.
```

Security and bug fixes MAY only be applied to the latest minor version.
We are committed to making it feasible for end users to stay up to date with the latest version of the OpenTelemetry SDK.

### Language version support

> Each language implementation SHOULD define

Reviewer 4: Could you format the paragraph more densely, without random breaks?

Author: It is not random. I tried to separate it by clauses. I find shorter lines easier to analyze when reviewing  diffs or when adding suggestions.

Reviewer 6: @pellared I would suggest that we pick a way that can be easily supported by some auto-formatting tool.

@yurishkuro please let us know if you think is a blocking issue or not (I personally feel it's not, although I agree that it should be more dense).

Reviewer 6: @yurishkuro I'll go ahead and merge this PR. If we feel strong about the wrapping, it can be a simple follow up editorial PR.

how the removal of a supported language/runtime version
affects its versioning.
As a rule of thumb,

> it SHOULD take the recommended approach in the given ecosystem

Reviewer 4: ```suggestion
it SHOULD follow the conventions in the given ecosystem.
```
